Mage.Sets/src/mage/cards/a/AutumnWillow.java

```diff
@@ -0,0 +1,110 @@
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.a;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.AsThoughEffectImpl;
+import mage.abilities.keyword.ShroudAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.TargetPlayer;
+
+/**
+ *
+ * @author L_J
+ */
+public class AutumnWillow extends CardImpl {
+
+    public AutumnWillow(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{G}{G}");
+        addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.AVATAR);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Shroud
+        this.addAbility(ShroudAbility.getInstance());
+
+        // {G}: Until end of turn, Autumn Willow can be the target of spells and abilities controlled by target player as though it didn't have shroud.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new AutumnWillowEffect(), new ManaCostsImpl("{G}"));
+        ability.addTarget(new TargetPlayer());
+        this.addAbility(ability);
+    }
+
+    public AutumnWillow(final AutumnWillow card) {
+        super(card);
+    }
+
+    @Override
+    public AutumnWillow copy() {
+        return new AutumnWillow(this);
+    }
+}
+
+class AutumnWillowEffect extends AsThoughEffectImpl {
+
+    public AutumnWillowEffect() {
+        super(AsThoughEffectType.SHROUD, Duration.EndOfTurn, Outcome.Benefit);
+        staticText = "Until end of turn, Autumn Willow can be the target of spells and abilities controlled by target player as though it didn't have shroud";
+    }
+
+    public AutumnWillowEffect(final AutumnWillowEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public AutumnWillowEffect copy() {
+        return new AutumnWillowEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (affectedControllerId.equals(source.getFirstTarget())) {
+            Permanent creature = game.getPermanent(sourceId);
+            if (creature != null) {
+                if (sourceId == source.getSourceId()) {
```

Comment:
this should be an .equals to be safe.

Reply:
👍 
True, I remember some longer debugging session to find out about such problems.